allauth/account/adapter.py

```diff
@@ -752,6 +752,9 @@ def get_reauthentication_methods(self, user):
                 )
         return ret
 
+    def get_email_verification_method(self, email=None):
```

Comment:
```suggestion
    def get_email_verification_method(self, login):
```

Now that we have a `Login` class representing the current login, I think it is more useful to base this method on an instance of that. See:

https://github.com/pennersr/django-allauth/blob/main/allauth/account/utils.py#L158

So, for that to happen, I think we need to:

- For every case where the default value (`email_verification=app_settings.EMAIL_VERIFICATION`) is passed, pass `None`. Then, inside the `__init__` of `Login`, when the email verification is not explicitly set to a value, call this adapter method.
-

Reply:
Thank you for reviewing I'll work on it.

Reply:
In assess_unique_email method https://github.com/pennersr/django-allauth/blob/main/allauth/account/utils.py#L496

We only have access to email not the Login instance as it is called before we complete signup. 
Is it a good idea to keep both login and email ?

Reply:
Would it work if we change the signature of `assess_unique_email` to include a `email_verification` parameter?

Reply:
How will it solve the problem in the sign up workflow ? Still login instance is not created before assess_unique_email method is invoked.

I think in the get_email_verification_method, if we have **kwargs, which will most likely contain, user fields (username, password, ...). Then developer overriding this method can make email_verification decision based on these field values. This will be beneficial, in case of custom sign up / login form (e-g add a field to determine email_verification method).

In the sign up and login workflow we can pass form fields and their values.